website/source/docs/platform/k8s/service-sync.html.md

```diff
@@ -0,0 +1,274 @@
+balancer, only one service instance will be registered with Consul rather
+than registering each individual pod endpoint.
+
+#### External IPs
+
+Any service type may specify an
+"[external IP](https://kubernetes.io/docs/concepts/services-networking/service/#external-ips)"
+configuration. The external IP must be configured by some other system, but
+any service discovery will resolve to this set of IP addresses rather than a
+virtual IP.
+
+If an external IP list is present, a service instance in Consul will be created
+for each external IP. It is assumed that if an external IP is present that it
+is routable and configured by some other system.
+
+### Sync Enable/Disable
+
+By default, all valid services (as explained above) are synced. This default
+can be changed as configuration to the sync process. Syncing can also be
+explicitly enabled or disabled using an annotation:
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-service
+  annotations:
+    "consul.hashicorp.com/service-sync": false
+```
+
+### Service Name
+
+When a Kubernetes service is synced to Consul, the name of the service in Consul
+by default will be the value of the "name" metadata on that Kubernetes service.
+This makes it so that service sync works with zero configuration changes.
+This can be overridden using an annotation to specify the Consul service name:
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-service
+  annotations:
+    "consul.hashicorp.com/service-name": my-consul-service
+```
+
+**If a conflicting service name exists in Consul,** the sync program will
+register additional instances to that same service. Therefore, services inside
+and outside of Kubernetes should have different names unless you want either
+side to potentially connect. This default behavior also enables gracefully
+transitioning a service from outside of K8S to inside, and vice versa.
+
+### Service Ports
+
+When syncing the Kubernetes service to Consul, the Consul service port will be
+the first defined port in the service. Additionally, all ports will be
+registered in the service instance metadata with the key "port-X" where X is
+the name of the port and the value is the externally accessible port.
+
+The default service port can be overridden using an annotation:
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-service
+  annotations:
+    "consul.hashicorp.com/service-port": "http"
+```
+
+The annotation value may a name of a port (recommended) or an exact port value.
+
+### Service Tags
+
+A service registered in Consul from K8S will always have the tag "k8s" added
+to it. Additional tags can be specified with a comma-separated annotation value
+as shown below. This will also automatically include the "k8s" tag which can't
+be disabled.
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-service
+  annotations:
+    "consul.hashicorp.com/service-tags": "primary,foo"
```

Comment:
When including multiple consul service tags here, is this the only formatting option for the tags to be correctly parsed? If I accidentally add spaces between tags or put them in brackets, will it cause problems?

Reply:
The system is robust enough to trim the whitespace, but I just updated the docs to explicitly state no whitespace just so that folks lean towards that.